pkg/api/meta/interfaces.go

```diff
@@ -40,6 +40,8 @@ type Interface interface {
 	SetNamespace(namespace string)
 	Name() string
 	SetName(name string)
+	GenerateName() string
```

Comment:
+1 the accessor made more sense back when I added it; it's pretty heavyweight now.